### ISSUE TITLE
[3.x] specify default page overwrited by `rewind` method

### ISSUE THREAD

**Issue by kichetof** [Sammyjo20/Saloon#307]
Hi Sam,

Thanks for your great package and this new v3!! 🔥

I try to upgrade v2 paginator and I was lose when I need to specify default page _(one of my api start at page=0)_.
I define the property `$page=0` on my anonymous class, but it still get page=1 on request:

```php
    public function paginate(Request $request): Paginator
    {
        return new class(connector: $this, request: $request) extends PagedPaginator
        {
            /**
             * Internal Marker For The Current Page
             */
            protected int $page = 0;

            // ... everything else
        }
    }
```

As `rewind` method is called on `count`, `$page` is reverted to [1](https://github.com/saloonphp/pagination-plugin/blob/v2/src/Paginator.php#L202).

Current workarround is to define `onRewind` method and overwrite `$page`

```php
public function paginate(Request $request): Paginator
    {
        return new class(connector: $this, request: $request) extends PagedPaginator
        {
            /**
             * Internal Marker For The Current Page
             */
            protected int $page = 0;

            /**
             * Apply additional logic on rewind
             *
             * This may be resetting specific variables on the paginator classes
             */
            protected function onRewind(): void
            {
                $this->page = 0;
            }
        };
    }
```

Maybe you have a better way to set default first page ? (on v2 it was: `$paginator->setCurrentPage(0)`)

_Probably define a new property `protected int $startPage = 1` (or 0 in my case) and use it in `rewind` : `$this->page = $this->startPage`?_

Thanks again for your excellent work!
Cheers!

**Comment by Sammyjo20** [comment]
Hey @kichetof thank you for the very kind words, it's really appreciated 🤠

Ah this is a really good point - I've gone ahead and taken your feedback and released v2.1.0 of the `pagination-plugin` which introduces a new `$startPage` property and `->setStartPage()` method on the paginator. I hope this is what you need? Please let me know if not.

https://github.com/saloonphp/pagination-plugin/pull/9/files

**Comment by kichetof** [comment]
Hi @Sammyjo20 

You're too fast! It works like a charm, many thanks! It's more readable to have this property instead of "cracking" onRewind 🚀

Thanks again for your great package! 🤠
